File: android/src/main/kotlin/com/emarsys/emarsys_sdk/command/geofence/GeofenceEnableCommand.kt
package com.emarsys.emarsys_sdk.command.geofence

import com.emarsys.emarsys_sdk.command.EmarsysCommand
import com.emarsys.emarsys_sdk.command.ResultCallback

class GeofenceEnableCommand : EmarsysCommand {
    override fun execute(parameters: Map<String, Any?>?, resultCallback: ResultCallback) {

    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        return true
    }

    override fun hashCode(): Int {
        return javaClass.hashCode()
    }
}

File: ios/Classes/Wrapper/Commands/Geofence/GeofenceEnableCommand.swift
//
//  Created by Emarsys on 2021. 08. 25..
//

import Foundation
import EmarsysSDK

class GeofenceEnableCommand: EmarsysCommandProtocol {
    
    func execute(arguments: [String : Any]?, resultCallback: @escaping ResultCallback) {
        Emarsys.geofence.enable()
        resultCallback(["success": true])
    }
}
